libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -1272,23 +1272,30 @@ string YulUtilFunctions::storageByteArrayPopFunction(ArrayType const& _type)
 	});
 }
 
-string YulUtilFunctions::storageArrayPushFunction(ArrayType const& _type)
+string YulUtilFunctions::storageArrayPushFunction(ArrayType const& _type, TypePointer _fromType)
 {
 	solAssert(_type.location() == DataLocation::Storage, "");
 	solAssert(_type.isDynamicallySized(), "");
+	if (_fromType && _fromType->isValueType())
+		solUnimplementedAssert(*_fromType == *_type.baseType(), "");
+
+	string functionName =
+		string{"array_push_"} +
+		(_fromType ? ("from_" + _fromType->identifier() + "_") : "" ) +
+		"to_" +
```

Comment:
```suggestion
		(_fromType ? ("from_" + _fromType->identifier() + "_to_") : "" ) +
```

Reply:
Since I set _fromType at the beginning, I added this regardless if that is fine